eng/testing/linker/SupportFiles/Directory.Build.targets

```diff
@@ -0,0 +1,25 @@
+<Project>
+  
+  <PropertyGroup>
+    <!-- Used to silence the warning caused by the workaround for https://github.com/dotnet/runtime/issues/81382 -->
+    <SuppressGenerateILCompilerExplicitPackageReferenceWarning>true</SuppressGenerateILCompilerExplicitPackageReferenceWarning>
+  </PropertyGroup>
+
+  <!-- needed to reference a specific version of NetCoreApp. Workaround https://github.com/dotnet/runtime/issues/81382 -->
+  <ItemGroup>
+    <FrameworkReference Update="Microsoft.NETCore.App"
+                        RuntimeFrameworkVersion="$(MicrosoftNETCoreAppRuntimeVersion)" />
+
+    <PackageReference Include="Microsoft.Dotnet.ILCompiler"
+                      Version="$(MicrosoftNETCoreAppRuntimeVersion)" />
+  </ItemGroup>
+
+  <!--
+  Reference the Microsoft.AspNetCore.App shared framework assemblies with ProjectReference.
+  This allows for easy dev workflow. Just need to update the product code, and re-run the tests.
+  -->
+  <ItemGroup>
```

Comment:
Did app.ref not work?

Reply:
It works, but it is a bit heavier than this approach.

For example, the "incremental" run of `.\.dotnet\dotnet.exe test .\src\DefaultBuilder\test\Microsoft.AspNetCore.NativeAotTests\` takes ~15s using app.ref and takes ~10s with this approach.

We can always switch this as we go forward if we need to. But for now I'd rather do the minimal.